### PR TITLE
Feature/add exercices questions order

### PR DESCRIPTION
**Pull request by NansD**
This branch implements the backend for allowing to update the order of an exercice or a question.
It takes inspiration from what M. Bourdeaud'huy did for our exam on Backend & Frontend development. 
(this kind of feature is a nightmare to test with postman haha)

I updated the routes for updating an exercice or a question.
I chose to requery the database to send the *effectively updated* document. This is not great in terms of performance, but I haven't found any other solution for the moment.
If we want to gain in performance, we can do just like before and send the `req.body` back to the user

### CODE REVIEW

**Review by FelixDelcourt — APPROVED** [review]

File: backend/src/services/exercice/data.js [hunk 1]
```diff
@@ -3,12 +3,38 @@ const Exercice = require('./model');
...
+
+
+  /**
+   * Updates the order of the other exercices that are contained
+   * in the same exam if it is needed
+   * @param {Exam} exam the parent exam that contains the exercices
+   * @param {Exercice} exercice the exercice that's being dragged and dropped
+   * @param {number} newExerciceOrder the new order of the exercice that's being dragged and dropped
+   */
+  async updateOrderOfOtherExercicesIfNeeded(exam, exercice, newExerciceOrder) {
```
Comment (ValentinChion) [inline backend/src/services/exercice/data.js]: Why not just call it updateOrder or something like that ? We can comment or put somewhere that it's updating only if needed

**Review by ValentinChion — APPROVED** [review]

File: backend/src/services/exercice/data.js (hunk 1 above)
Comment (NansD) [reply backend/src/services/exercice/data.js]: I wanted the name of the function to be very explicit, but yeah I can change if you prefer
@FelixDelcourt @bdebever @ArthurWirbel what do you prefer ?

Comment (ValentinChion) [reply backend/src/services/exercice/data.js]: A bit "overkill" as startupers would say :rocket:

Comment (NansD) [reply backend/src/services/exercice/data.js]: I changed the name 👍

**Pull request merged**